SSL_OP_NO_COMPRESSION not declared when building with OpenSSL 0.9.8k

Attempting to build ZNC 1.5 from git (commit id c8f49e1da02049d369eff5a1659fdd2314525ebd) on Ubuntu 10.04 Lucid Lynx I got the following error:

```
Building core object Csocket...
src/Csocket.cpp: In member function ?bool Csock::ConfigureCTXOptions(SSL_CTX*)?:
src/Csocket.cpp:1420:19: error: ?SSL_OP_NO_COMPRESSION? was not declared in this scope
    uCTXOptions |= SSL_OP_NO_COMPRESSION;
                   ^
make: *** [src/Csocket.o] Error 1
```

Some googling suggests that `SSL_OP_NO_COMPRESSION` was added in OpenSSL 0.9.9. I've worked around by adding

```
#ifndef SSL_OP_NO_COMPRESSION
#define SSL_OP_NO_COMPRESSION 0
#endif
```

to Csocket.c, which allows the file to compile, however I'm not sure if this is masking other potential problems related to using an outdated OpenSSL.

I suggest that if ZNC actually requires a newer version of OpenSSL then this should be tested for in the `configure` script, and if not, then my workaround should be added to allow it to build on older distros.

--- a/src/Csocket.cpp
+++ b/src/Csocket.cpp
@@ -1416,8 +1416,10 @@ bool Csock::ConfigureCTXOptions( SSL_CTX * pCTX )
 				uCTXOptions |= SSL_OP_NO_TLSv1_2;
 #endif /* SSL_OP_NO_TLSv1_2 */
 		}
+#ifdef SSL_OP_NO_COMPRESSION
 		if( m_bNoSSLCompression )
 			uCTXOptions |= SSL_OP_NO_COMPRESSION;
+#endif /* SSL_OP_NO_COMPRESSION */
 		if( uCTXOptions )
 			SSL_CTX_set_options( pCTX, uCTXOptions );
 	}

